src/lib.rs

```diff
@@ -726,26 +804,44 @@ impl Parse for Params {
 
         let mut debug = true;
         let mut default = true;
+        let mut endianness = Endianness::Little;
 
         // try parse additional args
         while <Token![,]>::parse(input).is_ok() {
             let ident = Ident::parse(input)?;
             <Token![=]>::parse(input)?;
-            let value = syn::LitBool::parse(input)?.value;
-            if ident == "debug" {
-                debug = value;
-            } else if ident == "default" {
-                default = value;
-            } else {
-                return Err(syn::Error::new(ident.span(), "unknown argument"));
-            }
+            match ident.to_string().as_str() {
```

Comment:
In this case, the conversion to a string leads to a memory allocation, which can take a long time. So two/three sting comparisons in an if-else chain might be faster. But a match looks better and more idiomatic, so I'm fine with both.